File: programmers/여행경로/seonghyeon.py
from collections import defaultdict
from copy import deepcopy

def solution(tickets):
    graph = defaultdict(list)
    for depart, dest in tickets:
        graph[depart].append(dest)

    for key in graph:
        graph[key].sort(reverse=True)

    count = len(tickets)
    answer = []

    def dfs(curr, route, graph):
        if len(route) ==count + 1:
            return route

        if not graph[curr]:
            return None

        for i in range(len(graph[curr]) - 1, -1, -1):
            next_graph = deepcopy(graph)
            dest = next_graph[curr].pop(i)

            new_route = dfs(dest, route + [dest], next_graph)
            if new_route:
                return new_route

        return None

    return dfs("ICN", ["ICN"], graph)

File: programmers/숫자 게임/seonghyeon.java
import java.util.Arrays;

class Solution {
    public int solution(int[] A, int[] B) {
        Arrays.sort(A);
        Arrays.sort(B);
        
        int answer = 0;
        int idx = 0;
        
        for (int i = 0; i < B.length; i++) {
            if (B[i] > A[idx]) {
                answer++;
                idx++;
            }
        }
        
        return answer;
    }
}
